cmplibrary/src/main/java/com/sourcepoint/gdpr_cmplibrary/SourcePointClient.java

```diff
@@ -2,112 +2,75 @@
-        public void onFailure(int statusCode, Header[] headers, String responseString, Throwable throwable) {
-            Log.d(LOG_TAG, "Failed to load resource " + url + " due to " + statusCode + ": " + responseString);
-            onLoadComplete.onFailure(new ConsentLibException(throwable.getMessage()));
-        }
-    }
-
-    void setHttpDummy(AsyncHttpClient httpClient) {
-        http = httpClient;
-    }
-
-    SourcePointClient(
-            int accountID,
-            String property,
-            int propertyId,
-            boolean isStagingCampaign,
-            boolean isStaging,
-            String targetingParams,
-            String authId
-    ) {
-        this.isStagingCampaign = isStagingCampaign;
-        this.isStaging = isStaging;
-        this.accountId = accountID;
-        this.propertyId = propertyId;
-        this.property = property;
-        this.targetingParams = targetingParams;
-        this.authId = authId;
+    SourcePointClient(OkHttpClient httpClient, SourcePointClientConfig config) {
+        this.httpClient = httpClient;
+        this.config = config;
     }
 
     void getMessage(boolean isNative, String consentUUID, String meta, String euconsent, GDPRConsentLib.OnLoadComplete onLoadComplete) throws ConsentLibException {
         String url = messageUrl(isNative);
         Log.d(LOG_TAG, "Getting message from: " + url);
-        try {
-            http.post(null, url, new StringEntity(messageParams(consentUUID, meta, euconsent).toString()), "application/json", new ResponseHandler(url, onLoadComplete) {
-                @Override
-                public void onSuccess(int statusCode, Header[] headers, JSONObject response) {
-                    Log.i(LOG_TAG, response.toString());
-                    onLoadComplete.onSuccess(response);
-                }
 
-                @Override
-                public void onFailure(int statusCode, Header[] headers, String responseString, Throwable throwable) {
-                    Log.d(LOG_TAG, "Failed to load resource " + url + " due to " + statusCode + ": " + responseString);
-                    onLoadComplete.onFailure(new ConsentLibException(throwable.getMessage()));
-                }
+        final MediaType mediaType= MediaType.parse("application/json");
+        RequestBody body = RequestBody.create(mediaType, messageParams(consentUUID, meta, euconsent).toString());
+
+        Request request = new Request.Builder().url(url).post(body)
+                .header("Accept", "application/json")
+                .header("Content-Type", "application/json")
+                .build();
+
+        httpClient.newCall(request).enqueue(new Callback() {
+            @Override
+            public void onFailure(Call call, IOException e) {
+                Log.d(LOG_TAG, "Failed to load resource " + url + " due to " +   "url load failure :  " + e.getMessage());
+                onLoadComplete.onFailure(new ConsentLibException(e, "Fail to send consent to: " + url));
```

Comment:
```java
onLoadComplete.onFailure(new ConsentLibException(e, "Fail to send consent to: " + url));
```
Maybe the reason `Fail to send consent to` is not accurate here as we're not sending consent but getting the message?

Reply:
This might be a _copy n paste_ mistake as the code seems similar from below.

Reply:
yeap, gonna change it 👍